internal/pkg/task/logs.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package task
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/aws/cloudwatchlogs"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+	"io"
+	"time"
+)
+
+const (
+	numCWLogsCallsPerRound = 10
+)
+
+// TasksDescriber describes ECS tasks.
+type TasksDescriber interface {
+	DescribeTasks(cluster string, taskARNs []string) ([]*ecs.Task, error)
+}
+
+// TaskEventsLogger gets a log group's log events.
+type TaskEventsLogger interface {
+	TaskLogEvents(logGroupName string,
+			streamLastEventTime map[string]int64,
+			opts ...cloudwatchlogs.GetLogEventsOpts) (*cloudwatchlogs.LogEventsOutput, error)
+}
+
+// EventsWriter represents a writer that writes tasks' log events to a writer.
+type EventsWriter struct {
+	GroupName string
+	Tasks     []*Task
+
+	Writer io.Writer
+	EventsLogger TaskEventsLogger
+	Describer TasksDescriber
+
+	// Fields that are private that get modified each time
+	lastEventTimestampByLogGroup map[string]int64
+	runningTasks []*Task
+}
+
+
+// WriteEventsUntilStopped writes tasks' events to a writer until all tasks have stopped.
+func (ew *EventsWriter) WriteEventsUntilStopped() error {
+	startTime := earliestStartTime(ew.Tasks)
+	ew.runningTasks = ew.Tasks
+	ew.lastEventTimestampByLogGroup = make(map[string]int64)
+
+	for {
+		for i := 0; i < numCWLogsCallsPerRound; i++ {
+			if err := ew.writeEvents(cloudwatchlogs.WithStartTime(aws.TimeUnixMilli(startTime))); err != nil {
+				return err
+			}
+			time.Sleep(cloudwatchlogs.SleepDuration)
+		}
+		stopped, err := ew.areTasksStopped()
+		if err != nil {
+			return err
+		}
+		if stopped {
+			return nil
+		}
+	}
+}
+
+// TODO: move this to a different package because this is not a task-specific method
```

Comment:
don't worry about this, i'll update this as well when refactoring the cwlogs pkg. ideally `TaskLogEvents` should accept an io writer as a param then take care of outputting.

Reply:
coool!!